### PR TITLE
map enhancement, add url parameter icon #507

### PR DESCRIPTION
**Pull request by astroidex**
* this pull request is for branch release/3.0.5
* see ticket https://github.com/mapbender/mapbender/issues/507

### CODE REVIEW

File: src/Mapbender/CoreBundle/Element/Map.php [hunk 1]
```diff
@@ -162,7 +162,12 @@ public function getConfiguration()
                     'x' => floatval($point[0]),
                     'y' => floatval($point[1]),
                     'label' => isset($poi['label']) ? htmlentities($poi['label']) : null,
-                    'scale' => isset($poi['scale']) ? intval($poi['scale']) : null
+                    'scale' => isset($poi['scale']) ? intval($poi['scale']) : null,
+                    'icon' => isset($poi['icon']) ? $poi['icon'] : null,
```
Comment (eSlider) [inline src/Mapbender/CoreBundle/Element/Map.php]: @LazerTiberius, please take a look on this.

**Comment by LazerTiberius** [comment]
I tried different possible exploits but without success. OL2 seems to do a good job here. The only problem I see is the possibility to load any file with the parameter. But this is more a problem with the general design. Maybe we should consider to only allow predefined icons and only use a name/id as a parameter?

**Comment by eSlider** [comment]
@LazerTiberius, if this good, please mark as approved. thx!

**Comment by LazerTiberius** [comment]
> But this is more a problem with the general design. Maybe we should consider to only allow predefined icons and only use a name/id as a parameter?

There is still an open question from me?

**Comment by AxxL** [comment]
Since this is a feature the pull-request has to be done against the 3.0.7 branch.

@eSlider @LazerTiberius I don't think using anything than predefined icons is a security thread and should not be put into mapbender without validating the concerns. If remote icons should be displayed, the core must be hardened to avoid loading remote JS-code or anything else except for an icon.

 @astroidex @eSlider As this is a feature, it must be a pull request against 3.0.7.

**Pull request closed without merging**